build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/biomeTasks.ts

```diff
@@ -39,41 +37,26 @@ export class BiomeTask extends LeafWithFileStatDoneFileTask {
 	}
 
 	/**
-	 * Includes all files in the task's package directory and any biome config files in the directory tree. Files ignored
-	 * by git are excluded.
+	 * Includes all files in the the task's package directory that Biome would format and any Biome config files that
+	 * apply to the directory. Files ignored by git are excluded.
 	 */
 	protected async getInputFiles(): Promise<string[]> {
-		const repoRoot = await this.repoRoot;
 		const gitRepo = await this.gitRepo;
+		// Absolute paths to files that would be formatted by biome.
+		const files = await getBiomeFormattedFiles(this.node.pkg.directory, gitRepo);
+		const configPath = await getClosestBiomeConfigPath(this.node.pkg.directory);
```

Comment:
In these lines, getClosestBiomeConfigPath is called twice, right?
Then configs are loaded multiple times to resolve getAllBiomConfigPaths below.
Seems like these can coordinate to be efficient. Seems like a good opportunity to have a helper class that loads the things once and can answer the questions using what it already has. Otherwise, big helper that function that returns all of the parts in an object `{ formattedFiles: string[]; configFiles: string[] }`.

Reply:
The class is a nice idea as a supplement to the free functions. I did a basic implementation in the latest iteration but I need to do a bit more refactoring to get rid of the duplicate work in some of the free functions.